### PR TITLE
Upgrade Robolectric to 4.7

### PR DESCRIPTION
**Pull request by TranceLove**
## Description
Additionally, inside SshConnectionPool, eliminated the CountdownLatch when waiting for converting PEM to private key AsyncTask; instead it became a blocking call. Should not hurt performance by much, but with simpler code and most importantly, work around Robolectric's default PAUSED LooperMode, which doesn't work well with code containing CountdownLatches.
  
#### Manual tests
- [x] Done  
  
- Device: Pixel 2 emulator
- OS: Android 11
SSH connection with key authentication should not be affected by the refactoring to SshConnectionPool.

#### Build tasks success  
Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ssh/PemToKeyPairTask.kt [hunk 1]
```diff
@@ -65,7 +65,7 @@ import java.security.KeyPair
  */
 class PemToKeyPairTask(
     private val pemFile: ByteArray,
-    private val callback: AsyncTaskResult.Callback<KeyPair?>
+    private val callback: AsyncTaskResult.Callback<KeyPair?>?
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ssh/PemToKeyPairTask.kt:68]: Why is this nullable? Is it ever acually null?

**Comment by EmmanuelMess** [comment]
I am getting a lot of:
```
java.io.IOException: unrecognised object: OPENSSH PRIVATE KEY
	at org.bouncycastle.openssl.PEMParser.readObject(Unknown Source)
	at com.amaze.filemanager.asynchronous.asynctasks.ssh.PemToKeyPairTask$JcaPemToKeyPairConverter.throwingConvert(PemToKeyPairTask.kt:201)
	at com.amaze.filemanager.asynchronous.asynctasks.ssh.PemToKeyPairTask$PemToKeyPairConverter.convert(PemToKeyPairTask.kt:188)
	at com.amaze.filemanager.asynchronous.asynctasks.ssh.PemToKeyPairTask.doInBackground(PemToKeyPairTask.kt:95)
	at com.amaze.filemanager.asynchronous.asynctasks.ssh.PemToKeyPairTask.doInBackground(PemToKeyPairTask.kt:66)
	at android.os.AsyncTask$2.$$robo$$android_os_AsyncTask_2$call(AsyncTask.java:287)
	at android.os.AsyncTask$2.call(AsyncTask.java)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at android.os.AsyncTask$SerialExecutor$1.$$robo$$android_os_AsyncTask_SerialExecutor_1$run(AsyncTask.java:230)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

In the logs after running tests.

**Comment by TranceLove** [comment]
@EmmanuelMess The design is to let incoming key data go through a series of PEM decoders, until one can decode the key format.

We will never know the PEM data is in which format, hence this design.

**Comment by TranceLove** [comment]
... and the second problem was due to missing ShadowCryptUtil. Fixed.

**Review by EmmanuelMess — APPROVED** [review]

**Review by VishalNehra — APPROVED** [review]

**Pull request merged**